qemu/tests/fwcfg64.py

```diff
@@ -0,0 +1,64 @@
+import logging
+
+from avocado.utils import process
+from virttest import env_process
+from virttest import error_context
+from virttest import utils_disk
+from provider import win_dump_utils
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Fwcfg64 basic function test:
+
+    1) Boot guest with -device vmcoreinfo.
+    2) Check the fwcfgg64 driver has been installed.
+    3) Run "dump-guest-memory -w memory.dmp" in qemu monitor.
+    4) Check the memory.dmp can be saved and the size is larger then 0Kb.
+    5) Check the dump file can be open with windb tools.
+    """
+    def set_vm_param():
+        """
+        Update vm mem and image size params to match fwcfg64 test needed.
+
+        :param params: Params object
+        """
+        host_free_mem = int(process.run(params["get_free_mem"],
+                                        shell=True).stdout_text)
+        host_avail_disk = int(process.run(params["get_avail_disk"],
+                                          shell=True).stdout_text)
+        if host_avail_disk < (host_free_mem / 1024) + 30:
```

Comment:
Would you help explain why `+30`? If it is guest mem, better get it from default `params['mem']` as it is possibly to be configured.

Reply:
I'll try to update it with params["image_size"]